RIOT-OS/RIOT
# cpu/esp32: enable puf_sram feature

### Contribution description

This pull request enables the `puf_sram` feature for ESP32 platforms.


### Testing procedure

Execute *tests/puf_sram* on a ESP32 board as explained in the test [README](https://github.com/RIOT-OS/RIOT/tree/master/tests/puf_sram#readme). I have used the `esp32-wroom-32` board. 

Run the example test script alike `python tests/example_test.py -n 100`. This produces 100 SRAM based seeds with a power-off time of 1 second. It should produce outputs similar to:

```
Iteration 0/100
805719428
Iteration 1/100
2758876560
Iteration 2/100
...
Iteration 99/100
3625304372
Number of seeds: 100       
Seed length    : 32 Bit   
Abs. Entropy   : 28.88 Bit   
Rel. Entropy   : 90.24 perc. 
```
**Note:** The reset button (EN pin) disables a voltage regulator which causes a hard reset. Hence, pressing reset on that board will **not** trigger the soft reset detection.

### Issues/PRs references

#17665
